### PR TITLE
Flashbang tweaks.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Meant as a "gentler" solution to the flashbang issue than #18512.

- Banger stun for at least 20% less by default (max stun is 16 from 20 seconds)
- Reduced range of the current "bang" effect by 2 tiles
- The "flash" effect is reduced to a simple Knockdown for 1 sec, from a 10 seconds stun if you're more than 5 tiles away and not a silicon, and a 2 seconds knockdown if you're more than 3 tiles away but less than 5 and not a silicon.
- The "bang" effect is reduced to a simple Knockdown for 2 sec if you're more than 3 tiles away, and is not a silicon, from a 4 seconds stun.

Basically, if you're more than 3 tiles away, the stun is capped to 2 seconds, instead of 14 previously, with then, by degrees of proximity, 6, 14, and 16 without any protection, 1 and 8 with simply ear protection, and 1, 4, 6 and 8 with simply eye protection.

**But why?**

To me, the problem with banger was more their ridiculous range and duration (14 seconds as a BASE, WIDESCREEN, for unprotected targets) rather than the ability to throw them with a 1 second duration, which was established as a feature (see old banger description box).

The numbers are temporary and can be further adjusted (= lowered) if need be.

I just don't believe outright **removing** the bangers is the solution, as they have legitimate uses. I believe the issue is to make them less overwhelming. The range is a way to address this.

I'll think of adding a way to get ghetto ear protection in a later PR.

:cl:
- tweak: Flashbang duration has been reduced by 20%, and now stuns carbons for a considerably shorter time at long range (more than 3 tiles). The cap for stunning at long range is 2 seconds, from 20 for carbons, and 8 seconds from 20 for silicons.

### CODE REVIEW

**Comment by Irradiation** [comment]
>Ghetto ear protection
Just like, use a ghetto space helmet man.

**Comment by iFlashYou** [comment]
It's simple, we just remove flashbangs

**Comment by SonixApache** [comment]
This is a decent alternative, worth testing out on live.

**Comment by ShiftyRail** [comment]
@iFlashYou I'd more than happy with a sleeptoxin smoke grenade alternative

**Comment by Irradiation** [comment]
It's simple, we just remove R&D and the research outpost.

**Comment by FatCat1978** [comment]
Stuffing crayons in the ears might work well as ghetto hearing protection

**Comment by Blithering** [comment]
Maybe have a blind/disorient/screenshake effect if you're far away but unprotected, rather than knockdown?

**Comment by ghost** [comment]
Nerfing flashbangs as a whole does not fix the issue of 1sec flashbangs being vastly superior to unmodified flashbangs. In fact after a nerf, people will want to set them to 1sec more to compensate.

**Comment by ShiftyRail** [comment]
@Unknown-as-Captain I don't see it as an issue, grenades are meant to be modified with a screwdriver. For flashbangs, it's just than 1sec (3sec if you have lags) is the superior choice in most cases.

The same way cleaner grenades/weedkiller grenades are set to a longer time.

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 1]
```diff
@@ -72,24 +81,32 @@
...
-			M.Knockdown(8)
+			M.Stun(4)
+			M.Knockdown(4)
 			M.ear_damage += rand(0, 3)
 			M.ear_deaf = max(M.ear_deaf,10)
 
 	else if(!ear_safety)
-		M.Stun(4)
-		M.Knockdown(4)
+		if (issilicon(M))
```
Comment (jknpj) [inline code/game/objects/items/weapons/grenades/flashbang.dm:107]: is this, dare i say, biased?

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 2]
```diff
@@ -56,8 +56,17 @@
 //Flashing everyone
 	if(eye_safety < 1)
 		M.flash_eyes(visual = 1, affect_silicon = 1)
-		M.Stun(10)
-		M.Knockdown(10)
+		if (get_dist(M, T) <= 3)
+			M.Stun(8)
+			M.Knockdown(8)
+		else
+			if (issilicon(M))
```
Comment (jknpj) [inline code/game/objects/items/weapons/grenades/flashbang.dm:63]: Hmm....

Comment (SonixApache) [reply code/game/objects/items/weapons/grenades/flashbang.dm:63]: 🤔

**Comment by ShiftyRail** [comment]
@jknpj I completely assume this, in the sense that unlike carbons, sillies cannot be stunned by a taser, or tripped by bolas.

**Comment by jknpj** [comment]
that can be arranged

**Comment by Irradiation** [comment]
Arrange it first then we're talking.

**Comment by jknpj** [comment]
gotcha motherfucker

**Comment by nuklearcellphoneg** [comment]
The reason people don't like one second flash bangs is because they're instant so there's no way to defend against them, unless you want to be meta and power game stuff preemptively. They also involve zero skill, so when they get knocked out by them it's very, very lame. They're a security equivalent to ye olde chloral autoinjector or the para-silence sting.  People don't mind tasers because you can attempt to dodge tasers, and if you dodge a full clip, awesome. You feel like a badass and run away or something. The same can be kind of said for lasers, though they're more accurate, but ballistic guns have the same idea. What people do mind is the stun system we have, and the flash bang is a pretty good embodiment of some of the main things wrong with it. 

The changes only seem to buff the flash grenade from what I can tell. They're more accurate now with less friendly fire risk. The people getting flashed also can't utilize the 3-5 tile spacing... I mean, it's still going to land near you, stun you, boom. Game over. If you shoot a taser at me, it hits or misses. Okay, so you shoot more, or I do something. You throw the grenade in? There's absolutely nothing you can do.

Oh, you managed to happen to be in the 2 second stun zone? One second to step forward and see 'em stunned, then one second priming and throwing my second flash bang. :^)

If you really want to improve flash grenades, get rid of the stunning aspect completely and just implement a more robust blind/deaf period from the flash recipe on up, otherwise they either need to be removed or the one second insta-option needs to be redone, because nobody likes equipment like that to which there is no defense or counter for it except for the people who use said equipment... which would be Security.

**Comment by Irradiation** [comment]
"No defense or counter for it except for the people who use said equipment... which would be Security."
Or wear any variant of a hardsuit helmet that tickles your fancy, but yeah. 

I'm all up for bangers making banged people dizzy and/or confused for up to 5 seconds on top of the blind+deaf instad of stun+blind+deaf.

**Comment by Pathid** [comment]
@ShiftyRail Silicons do however get stunned by bolas. It's not a long stun (4 seconds), but certainly enough to run up on them and use a flash.

**Comment by nuklearcellphoneg** [comment]
>Or wear any variant of a hardsuit helmet that tickles your fancy, but yeah.

I'd argue hardsuit helmets shouldn't, but that's beside the point, so let me posit it this way. 

Sec Officer runs up on you, you're wanted, you have nothing. 

If he has only a taser and cuffs, you can beat him with your bare fists with a little know how and luck. There is a piece of equipment that can increase your chances, the Ablative. You don't need it to beat them, but it would be helpful- however you don't NEED the Ablative and for onlookers it's more impressive that you don't use it.

If a sec officer runs up with only a single one second flash bang and cuffs, you're 100% boned unless you already set yourself up(and let's face it even if you are, even with the above changes you're boned). You can't run away once you realize it's there, you can't dodge it... it's more imperative you get a hardsuit helmet to go against security than it is to wear an ablative to counter tasers and lasers.

>I'm all up for bangers making banged people dizzy and/or confused for up to 5 seconds on top of the blind+deaf instad of stun+blind+deaf.

I don't think dizzy/confused is necessary really, if the blind/deaf duration is long enough. Yackety saxing around tasers is by no means 100% luck, and being blind/deaf hampers your ability to dodge these greatly. 

To point back to my previous post, if someone gets dunked by an insta-flashbang, their thought afterwards is probably along the lines of that the grenade is at fault for being OP, they definitely don't think 'Oh man this is my fault I could have done X better just now' or 'Wow this sec officer sure is robust!!'

If all it did was blind/deaf for a period, and they got tased/killed during this, it would feel a lot more deserved, because they could have done better at running away while blind or what have you. If it made people dizzy/confused, it would again make people think, 'Well if that dumb grenade didn't make me lose control of my character'.

**Comment by SonixApache** [comment]
Causing `agony` would be an option alongside blinding and deafening.

**Comment by ShiftyRail** [comment]
@nuklearcellphoneg Bangers exist mostly to do crowd control, which is a situation that is warranted
I agree that they are too powerful, however, this is the intent of this PR : to make them less overwhelming and less "be all end all" solution, and rather a way to just control an area

The radius at which people are stunned is divided by more than 2 ; if you don't aim your banger at the person, they will be up in 2 seconds or less.

>It's hard to find protection against flashbangs

Sunglasses or hardsuits helmets exist

Also, in a world where cyborgs are not affected by tasers, and have infinite taser, zas immunity, and all, I believe it's fair to have a tool to screw them over

**Comment by nuklearcellphoneg** [comment]
Again, this doesn't change what people are disliking about them.  You can beat tasers and flash with ablative and sunglasses, but you don't need them, and because you do need them to beat the flashbang(if you are throwing, it instantly goes off, there's no way someone can react to it, and assuming the one who threw it did so as to put them in the 2 sec stun... they are instant grenades, you can instantly throw a second), you can't throw that flashbang back, and you can't run to a safe distance, you can't do anything except assume before anyone ever uses one that you need the protection from one.

Unless you remove the stun altogether, people will continue to feel the exact same way about them.

**Comment by Irradiation** [comment]
I'll check how much of shitcode flashbangs are regarding their AOE size, but in the meantime, we could look into lowering the throw speed of bangers so they aren't an instant throw directly under your feet.

**Comment by Tokanova** [comment]
lol there was literally a vote earlier for no nerfing, and now you're trying to do it again? gay

**Pull request merged**